### PR TITLE
UI 2.0 remove nginx container

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes: #DNE
# What's Changed

- Replace nginx reverse proxy with [CRA proxy](https://create-react-app.dev/docs/proxying-api-requests-in-development/) feature
- Replace `node:latest` docker image with `node:slim` (reduced filesize from 1GB to 250MB)

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: development/docker-compose.dev.yml [hunk 1]
```diff
@@ -32,17 +31,9 @@ services:
     user: node
     volumes:
       - ../nautobot/ui/:/opt/node/nautobot
+      - ../examples/:/opt/examples
```
Comment (glennmatthews) [inline development/docker-compose.dev.yml:34]: Curious why this is needed? Can we not pull the example plugin in from `/source`?

File: nautobot/ui/package.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
   "name": "nautobot",
   "version": "2.0.0-alpha.1",
   "private": true,
+  "proxy": "http://nautobot:8080",
```
Comment (glennmatthews) [inline nautobot/ui/package.json:5]: This feels like a strange place for this, though I'm definitely no expert here. What's the impact of this addition?

Comment (gsnider2195) [reply nautobot/ui/package.json:5]: This is what tells node to proxy any unknown routes to the backend nautobot container (so we can remove nginx)

File: development/docker-compose.dev.yml (hunk 1 above)
Comment (gsnider2195) [reply development/docker-compose.dev.yml:34]: We're using relative paths in our `jsconfig.json` so our path to the example plugin ui looks like `../../examples/ui/`. This is to support the same relative path in the node container.

**Review by bryanculver — APPROVED** [review]

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**